starknet_py/net/account/base_account.py

```diff
@@ -100,14 +101,16 @@ async def sign_declare_transaction(
         self,
         compiled_contract: str,
         *,
+        compiled_class_hash: Optional[int] = None,
         max_fee: Optional[int] = None,
         auto_estimate: bool = False,
-    ) -> Declare:
+    ) -> Union[Declare, DeclareV2]:
         """
         Create and sign declare transaction.
 
         :param compiled_contract: string containing compiled contract bytecode.
             Useful for reading compiled contract from a file.
+        :param compiled_class_hash: a class hash of the compiled contract used in the declare transaction.
```

Comment:
It might be useful to specify, what exactly the `compiled_contract` is - the output of Cairo1 compiler, containing Sierra bytecode.
And then we could also explain `compiled_class_hash` - the class_hash of the class containing Cairo Assembly, obtained by compiling Sierra Class using Sierra -> Casm compiler.

Reply:
I'll updated this description a bit.